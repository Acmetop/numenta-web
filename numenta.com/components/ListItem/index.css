/* Numenta company website source code. Copyright Â© 2016 Numenta.*/
/* Full details in LICENSE.txt, or contact us at <http://numenta.com>.*/
/* This program is free software: you can redistribute it and/or modify it under*/
/* the terms of the GNU Affero General Public License as published by the Free*/
/* Software Foundation, either version 3 of the License, or (at your option) any*/
/* later version. This program is distributed in the hope that it will be*/
/* useful, but WITHOUT ANY WARRANTY; without even the implied warranty of*/
/* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero*/
/* General Public License for more details. You should have received a copy of*/
/* the GNU Affero General Public License along with this program. If not, see*/
/* <https://www.gnu.org/licenses/agpl.html>.*/

._marker {
  composes: ml4 from 'tachyons';
  composes: mb2 from 'tachyons';
  composes: pl1 from 'tachyons';
}


.markerCircle {
  composes: _marker;
  list-style-type: circle;
}
.markerDecimal {
  composes: _marker;
  list-style-type: decimal;
}
.markerDisc {
  composes: _marker;
  list-style-type: disc;
}
.markerIcon {
  composes: list from 'tachyons';
}
.markerSquare {
  composes: _marker;
  list-style-type: square;
}

.copy {
  composes: lh-copy from 'tachyons';
  composes: measure from 'tachyons';
  composes: measure-wide-l from 'tachyons';
}

.li {
  /* */
}
