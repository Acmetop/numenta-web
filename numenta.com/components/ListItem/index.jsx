// Numenta company website source code. Copyright Â© 2016 Numenta.
// Full details in LICENSE.txt, or contact us at <http://numenta.com>.
// This program is free software: you can redistribute it and/or modify it under
// the terms of the GNU Affero General Public License as published by the Free
// Software Foundation, either version 3 of the License, or (at your option) any
// later version. This program is distributed in the hope that it will be
// useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
// General Public License for more details. You should have received a copy of
// the GNU Affero General Public License along with this program. If not, see
// <https://www.gnu.org/licenses/agpl.html>.

import classNames from 'classnames'
import React from 'react'

import styles from './index.css'

const markers = {
  circle: styles.markerCircle,
  decimal: styles.markerDecimal,
  disc: styles.markerDisc,
  icon: styles.markerIcon,
  square: styles.markerSquare,
}


/**
 *
 */
const ListItem = ({children, copy, marker}) => {
  const classes = [styles.li]

  if (copy) classes.push(styles.copy)
  if (marker) classes.push(markers[marker])

  return (
    <li className={classNames(...classes)}>
      {children}
    </li>
  )
}

ListItem.propTypes = {
  children: React.PropTypes.any.isRequired,
  copy: React.PropTypes.bool,
  marker: React.PropTypes.oneOf(Object.keys(markers)),
}

ListItem.defaultProps = {
  copy: true,
}

export default ListItem
