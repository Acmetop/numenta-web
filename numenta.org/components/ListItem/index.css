/**
 * Numenta.org HTM Community website source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2016 Numenta <http://numenta.com>
 */

._marker {
  composes: ml4 from 'tachyons';
  composes: mb2 from 'tachyons';
  composes: pl1 from 'tachyons';
}


.markerCircle {
  composes: _marker;
  list-style-type: circle;
}
.markerDecimal {
  composes: _marker;
  list-style-type: decimal;
}
.markerDisc {
  composes: _marker;
  list-style-type: disc;
}
.markerIcon {
  composes: list from 'tachyons';
}
.markerSquare {
  composes: _marker;
  list-style-type: square;
}

.copy {
  composes: lh-copy from 'tachyons';
  composes: measure from 'tachyons';
  composes: measure-wide-l from 'tachyons';
}

.li {
  /* */
}
