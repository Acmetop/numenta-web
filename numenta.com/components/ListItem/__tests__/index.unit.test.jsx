import React from 'react'
import renderer from 'react-test-renderer'

import ListItem from '../../ListItem'


describe('ListItem React component', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <ListItem>
        List Item 1
      </ListItem>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
