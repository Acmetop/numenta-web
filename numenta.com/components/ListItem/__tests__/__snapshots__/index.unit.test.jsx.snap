exports[`ListItem React component Renders correctly 1`] = `
<li
  className="li copy">
  List Item 1
</li>
`;
